exporters/otlp/otlp.go

```diff
@@ -59,10 +59,15 @@ type Exporter struct {
 var _ tracesdk.SpanExporter = (*Exporter)(nil)
 var _ metricsdk.Exporter = (*Exporter)(nil)
 
-func configureOptions(cfg *Config, opts ...ExporterOption) {
+func NewConfig(opts ...ExporterOption) Config {
```

Comment:
Should this remain unexported?

Reply:
The `Config` type is exported, so I figured the initializer should be as well.  Though it does seem that I need to add a docblock there.

Reply:
> The `Config` type is exported, so I figured the initializer should be as well. Though it does seem that I need to add a docblock there.

Good point, sounds reasonable to me. +1 on the docs as well :+1:

Reply:
All fields are unexported in `Config`, maybe we should also make the `Config` unexported? so we can remain this function unexported.

https://github.com/open-telemetry/opentelemetry-go/blob/c9726efd6075c1dccbafc5a17724f0299fd1998a/exporters/otlp/options.go#L66-L76

Reply:
Excellent point.  Without any exported fields I suppose there's no value in having an exported configuration struct.  The whole point would be for users to write custom options which would need to be able to manipulate fields in the struct.